File: social-service/src/main/java/com/eirsteir/coffeewithme/social/domain/friendship/FriendshipStatus.java
package com.eirsteir.coffeewithme.social.domain.friendship;

import com.eirsteir.coffeewithme.social.domain.RequestStatus;

public enum FriendshipStatus implements RequestStatus {
  REQUESTED(1),
  ACCEPTED(2),
  DECLINED(3),
  BLOCKED(4);

  private Integer value;

  FriendshipStatus(Integer value) {
    this.value = value;
  }

  public String getStatus() {
    return this.name();
  }

  public Integer getValue() {
    return value;
  }
}

File: social-service/src/main/kotlin/com/eirsteir/coffeewithme/social/domain/friendship/FriendshipStatus.kt
package com.eirsteir.coffeewithme.social.domain.friendship

import com.eirsteir.coffeewithme.social.domain.RequestStatus

enum class FriendshipStatus(private val value: Int) : RequestStatus {
    REQUESTED(1),
    ACCEPTED(2),
    DECLINED(3),
    BLOCKED(4);

    override fun getStatus() = name

    override fun getValue() = value
}